test/StrikeSelectionE2E.ts

```diff
@@ -0,0 +1,249 @@
+
+    it("reverts when not owner call", async function () {
+      await expect(
+        strikeSelection.connect(signer2).setStep(50)
+      ).to.be.revertedWith("Ownable: caller is not the owner");
+    });
+
+    it("sets the step", async function () {
+      await strikeSelection.connect(signer).setStep(50);
+      assert.equal(
+        (await strikeSelection.step()).toString(),
+        BigNumber.from(50)
+          .mul(BigNumber.from(10).pow(await wethPriceOracle.decimals()))
+          .toString()
+      );
+    });
+  });
+
+  describe("getStrikePrice", () => {
+    time.revertToSnapshotAfterEach();
+
+    let underlyingPrice: BigNumber;
+    let deltaAtUnderlying: BigNumber;
+    let expiryTimestamp: BigNumber;
+
+    beforeEach(async () => {
+      await updateVol();
+      underlyingPrice = await optionsPremiumPricer.getUnderlyingPrice();
+      underlyingPrice = underlyingPrice.sub(
+        BigNumber.from(underlyingPrice).mod(await strikeSelection.step())
+      );
+      expiryTimestamp = (await time.now()).add(WEEK);
+      deltaAtUnderlying = await optionsPremiumPricer.getOptionDelta(
+        underlyingPrice,
+        expiryTimestamp
+      );
+    });
+
+    it("reverts on timestamp being in the past", async function () {
+      const isPut = false;
+      await expect(
+        strikeSelection.getStrikePrice((await time.now()).sub(100), isPut)
+      ).to.be.revertedWith("Expiry must be in the future!");
+    });
+
+    it("gets the correct strike price given delta for calls", async function () {
+      const isPut = false;
+      const targetDelta = await strikeSelection.delta();
+      const [strikePrice, delta] = await strikeSelection.getStrikePrice(
+        expiryTimestamp,
+        isPut
+      );
+
+      assert.equal(
+        strikePrice.toString(),
+        underlyingPrice
+          .add(
+            deltaAtUnderlying
+              .sub(targetDelta)
+              .div(500)
+              .mul(100)
+              .mul(BigNumber.from(10).pow(await wethPriceOracle.decimals()))
+          )
+          .toString()
+      );
+      assert.isBelow(
+        parseInt(targetDelta.toString()),
+        parseInt(delta.toString())
+      );
+      assert.isAbove(
+        parseInt(targetDelta.toString()),
+        parseInt(
+          await optionsPremiumPricer.getOptionDelta(
+            strikePrice.add(await strikeSelection.step()),
+            expiryTimestamp
+          )
+        )
+      );
+    });
+
+    it("gets the correct strike price given delta for puts", async function () {
+      const isPut = true;
+      const targetDelta = await strikeSelection.delta();
+      const [strikePrice, delta] = await strikeSelection.getStrikePrice(
+        expiryTimestamp,
+        isPut
+      );
+
+      assert.equal(
+        strikePrice.toString(),
+        underlyingPrice
+          .sub(
+            deltaAtUnderlying
+              .sub(targetDelta)
+              .div(500)
+              .sub(3)
```

Comment:
shifting by 300 here. seems counterintuitive, but say spot is 1800. 0.1 delta for call will be 2300 (diff of 500) and 0.1 delta for put will be 1500 (diff of 300). it makes sense that they would either be 500 from spot or 300 from spot. but they are not
<img width="926" alt="Screen Shot 2021-06-25 at 1 57 53 PM" src="https://user-images.githubusercontent.com/11217839/123485316-b8735200-d5be-11eb-90ac-44c19aaa0ffa.png">



>